test/runtime/binarymemcache.go

```diff
@@ -0,0 +1,212 @@
+	runWithClient := func(code string) *helpers.CmdRes {
+		logger.Infof("Executing %s", code)
+		res := vm.ContainerExec("memcache-client", fmt.Sprintf(
+			"python -c 'import bmemcached; client = bmemcached.Client((\"%s:11211\", )); %s'", memcacheIP, code))
+		return res
+	}
+
+	BeforeAll(func() {
+
+		logger = log.WithFields(logrus.Fields{"testName": memcacheTestName})
+		vm = helpers.InitRuntimeHelper(helpers.Runtime, logger)
+
+		ExpectCiliumReady(vm)
+
+		containers("create")
+		epsReady := vm.WaitEndpointsReady()
+		Expect(epsReady).Should(BeTrue(), "Endpoints are not ready after timeout")
+	})
+
+	AfterEach(func() {
+		vm.PolicyDelAll()
+	})
+
+	AfterAll(func() {
+		containers("delete")
+	})
+
+	JustAfterEach(func() {
+		vm.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
+	})
+
+	AfterFailed(func() {
+		vm.ReportFailed("cilium policy get")
+	})
+
+	It("Tests basic memcache operation", func() {
+		key := "test1"
+		value := "value1"
+		By("Setting memcache value")
+		r := setKey(key, value)
+
+		r.ExpectSuccess("Unable to set")
+		Expect(r)
+
+		By("getting memcache value")
+		r = getKey(key)
+		r.ExpectSuccess("Unable to get")
+		Expect(r.GetStdOut()).Should(ContainSubstring(value))
+	})
+
+	It("Tests policy allowing all actions", func() {
+		_, err := vm.PolicyImportAndWait(vm.GetFullPath("Policies-memcache-binary-allow.json"), 300)
+		Expect(err).Should(BeNil())
+
+		key := "test2"
+		value := "value2"
+		By("Setting memcache value")
+		r := setKey(key, value)
+		r.ExpectSuccess("Unable to set")
+		Expect(r)
+
+		By("getting memcache value")
+		r = getKey(key)
+		r.ExpectSuccess("Unable to get")
+		Expect(r.GetStdOut()).Should(ContainSubstring(value))
+	})
+
+	It("Tests policy disallowing set action", func() {
+		By("Setting key before disallowing set")
+		keyBeforePolicy := "before"
+		valBeforePolicy := "beforeval"
+		r := setKey(keyBeforePolicy, valBeforePolicy)
+		r.ExpectSuccess("Unable to set")
+
+		By("Importing policy disallowing set")
+		_, err := vm.PolicyImportAndWait(vm.GetFullPath("Policies-memcache-binary-disallow-set.json"), 300)
+		Expect(err).Should(BeNil())
+
+		By("Trying to set new key")
+		key := "keyAfterPolicy"
+		value := "valAfterPolicy"
+
+		By("Setting memcache value")
+		r = setKey(key, value)
+		r.ExpectFail()
+
+		By("getting memcache value")
+		r = getKey(keyBeforePolicy)
+		r.ExpectSuccess("Unable to get key set before applying policy")
+		Expect(r.GetStdOut()).Should(ContainSubstring(valBeforePolicy))
+	})
+
+	It("Tests policy allowing actions only for key", func() {
+		By("Importing key-specific policy")
+		_, err := vm.PolicyImportAndWait(vm.GetFullPath("Policies-memcache-binary-allow-key.json"), 300)
+		Expect(err).Should(BeNil())
+
+		key := "allowed"
+		value := "value"
+		By("Setting memcache allowed key")
+		r := setKey(key, value)
+		r.ExpectSuccess("Unable to set allowed key")
+		By("Getting memcache allowed key")
+		r = getKey(key)
+		r.ExpectSuccess("Unable to get allowed key")
+
```

Comment:
I would add here a test from a different container and test that test fail, to make sure that the from is working correctly.

Reply:
@eloycoto I don't understand, can you elaborate on what this test would do?

Reply:
You always test from `memcache-client` to a given key, but we never test from other container, I would like to have a memcache-client2 that test the same key and make sure that fails.